src/components/NodeConnectionView.js

```diff
@@ -49,6 +51,25 @@ export default class NodeConnectionView extends Component<Props, State> {
         props.connectionError &&
         props.connectionError.startsWith('You need to stake'),
     }
+    this.resolveEns()
+  }
+
+  async resolveEns () {
+    const { ethNetwork } = this.props
+    const resolverContract = ResolverContract(ethNetwork)
+    const stakeContract = StakeContract(ethNetwork)
+    try {
+      const requiredStake = await stakeContract.requiredStake()
```

Comment:
When possible better use the following code to run calls in parallel rather than in series:
```js
const [requiredStake, stakeAddress, tokenAddress] = await Promise.all([
  stakeContract.requiredStake(),
  resolverContract.resolve(ENS_NAMES.stake[ethNetwork]),
  resolverContract.resolve(ENS_NAMES.token[ethNetwork]),
])
```